src/layout/List/ListComponent.tsx

```diff
@@ -55,50 +46,34 @@ export const ListComponent = ({
     (state) => state.dataListState.dataLists[id]?.paginationData?.totaltItemsCount || 0,
   );
 
-  const handleChange = ({ selectedValue }: ChangeProps) => {
+  const handleChange = ({ selectedValue: selectedValue }: ChangeProps<Record<string, string>>) => {
     for (const key in formData) {
       handleDataChange(selectedValue[key], { key: key });
     }
   };
 
-  const renderRow = (datalist) => {
-    const cells: JSX.Element[] = [];
-    for (const key of Object.keys(datalist)) {
-      cells.push(<TableCell key={`${key}_${datalist[key]}`}>{datalist[key]}</TableCell>);
-    }
-    return cells;
-  };
-
-  const renderHeaders = (headers) => {
-    const cell: JSX.Element[] = [];
-    for (const header of headers) {
-      if ((sortableColumns || []).includes(header)) {
-        cell.push(
-          <TableCell
-            onChange={handleSortChange}
-            sortKey={header}
-            key={header}
-            sortDirecton={sortColumn === header ? sortDirection : SortDirection.NotActive}
-          >
-            {getTextResourceAsString(header)}
-          </TableCell>,
-        );
-      } else {
-        cell.push(<TableCell key={header}>{getTextResourceAsString(header)}</TableCell>);
+  const selectedRow: Record<string, string> = React.useMemo(() => {
+    let matchRow: boolean[] = [];
+    for (const row of calculatedDataList) {
+      for (const key in formData) {
+        matchRow.push(formData[key] === row[key]);
+      }
+      if (!matchRow.includes(false)) {
+        return row;
```

Comment:
I might be misunderstanding, but if `formData` is not set here (or is an empty object), the first row will be selected, right?

Reply:
No, it is only returned a row if one is clicked. If formData is not set or is an empty object the function returnes {}. We made a function to figure out which row that was selected by comparing the formdata and the rows. We made a list with only booleans and if that list doesn´t include any false when the loop is finished it returns the selectedRow. The loop checks all the fields in formdata against the row and if one of the fields does not match false is added to the string. It is not the best solution, but we could not figure out any better way. It is just to make suggestions if you have any :)

Reply:
Yup, I just figured `matchRow` won't contain any `false`-values when there is nothing in `formData` to iterate. In that case, `row` is returned. I guess you tested this with a fresh instance?

My suggestion would just be to return early with `{}` if `formData` is not set/is empty, making that behaviour explicit. So, when reading the function, it would say:
1. If `formData` is not set/empty, return an empty object
2. If `formData` matches one of the rows, return that row
3. Otherwise, return an empty object

As it stands, 1 is missing from the code.